web/packages/teleterm/src/ui/Vnet/VnetConnectionItem.tsx

```diff
@@ -160,19 +164,38 @@ const VnetConnectionItemBase = forwardRef<
 
         {/* Buttons to the right. Negative margin to match buttons of other connections. */}
         <Flex gap={1} mr="-3px">
-          {props.showHelpButton && (
-            <ButtonIcon
-              as="a"
-              title="Open VNet documentation"
-              href="https://goteleport.com/docs/connect-your-client/vnet/"
-              target="_blank"
-              onClick={e => {
-                // Don't trigger ListItem's onClick.
-                e.stopPropagation();
-              }}
-            >
-              <icons.Question size={18} />
-            </ButtonIcon>
+          {props.showExtraRightButtons && (
+            <>
+              <ButtonIcon
+                as="a"
+                title="Open VNet documentation"
+                href="https://goteleport.com/docs/connect-your-client/vnet/"
+                target="_blank"
+                onClick={e => {
+                  // Don't trigger ListItem's onClick.
+                  e.stopPropagation();
+                }}
+              >
+                <icons.Question size={18} />
+              </ButtonIcon>
+
+              {isVnetDiagEnabled && (
+                <ButtonIcon
+                  title={
+                    status.value === 'running'
+                      ? 'Run diagnostics'
+                      : 'Cannot run diagnostics while VNet itself is not running'
```

Comment:
Nit: I'd shorten it slightly
```suggestion
                      : 'VNet must be running to run diagnostics'
````